ChiTech/ChiPhysics/FieldFunction/fieldfunction_vtkpwlc.cc

```diff
@@ -46,448 +39,90 @@ void chi_physics::FieldFunction::ExportToVTKPWLC(const std::string& base_name,
-                                  std::string(".vtu");
-
-  //============================================= Serial Output each piece
-  vtkXMLUnstructuredGridWriter* grid_writer =
-    vtkXMLUnstructuredGridWriter::New();
-
-  ugrid->GetCellData()->AddArray(matarray);
-  ugrid->GetCellData()->AddArray(pararray);
-  ugrid->GetPointData()->AddArray(phiarray);
-  ugrid->GetCellData()->AddArray(phiavgarray);
-
-  grid_writer->SetInputData(ugrid);
-  grid_writer->SetFileName(location_filename.c_str());
-
-  grid_writer->Write();
-
-  //============================================= Parallel summary file
-  if (chi_mpi.location_id == 0)
-  {
-      WritePVTU(base_filename, field_name);
-  }
-}
-
-
-
-//###################################################################
-/**Handles the PWLD version of a field function export to VTK.
- *
- * */
-void chi_physics::FieldFunction::ExportToVTKPWLCG(const std::string& base_name,
-                                                  const std::string& field_name)
-{
-  if (spatial_discretization->type !=
-      chi_math::SpatialDiscretizationType::PIECEWISE_LINEAR_CONTINUOUS)
-    throw std::invalid_argument(std::string(__PRETTY_FUNCTION__) +
-                                " Field function spatial discretization"
-                                " is not of type "
-                                " PIECEWISE_LINEAR_CONTINUOUS.");
-
-  auto& pwl_sdm = static_cast<SpatialDiscretization_PWLC&>(*spatial_discretization);
-
-  std::vector<std::vector<double>>    d_nodes;
-
-  auto points = vtkSmartPointer<vtkPoints>::New();
-
-  auto& ff_uk = this->unknown_manager.unknowns[ref_variable];
-
-  //============================================= Init grid and material name
-  vtkUnstructuredGrid* ugrid;
-  vtkIntArray*      matarray;
-  vtkIntArray*      pararray;
-  std::vector<vtkDoubleArray*>   phiarray(ff_uk.num_components);
-  std::vector<vtkDoubleArray*>   phiavgarray(ff_uk.num_components);
-
-  ugrid    = vtkUnstructuredGrid::New();
-  matarray = vtkIntArray::New();
-  matarray->SetName("Material");
-  pararray = vtkIntArray::New();
-  pararray->SetName("Partition");
-
-  for (int g=0; g < ff_uk.num_components; g++)
-  {
-    char group_text[100];
-    sprintf(group_text,"%03d",g);
-    phiarray[g]    = vtkDoubleArray::New();
-    phiavgarray[g] = vtkDoubleArray::New();
-
-    phiarray[g]   ->SetName((field_name +
-                             std::string("_g") +
-                             std::string(group_text)).c_str());
-    phiavgarray[g]->SetName((field_name +
-                             std::string("_g") +
-                             std::string(group_text) +
-                             std::string("_avg")).c_str());
-  }
-
   //======================================== Precreate nodes to map
-
   std::vector<std::tuple<uint64_t,uint,uint>> cell_node_component_tuples;
 
-  for (unsigned int g=0; g < ff_uk.num_components; g++)
+  for (unsigned int c=0; c < num_components; ++c)
```

Comment:
++c or c++. what's the difference again and is it consistent with the other loops in this PR?

Reply:
`++c` increments the value and returns the new value (incremented). `c++` increments the value and returns the old value (un-incremented). Doesn't really matter for integral values but `++c` in loops is good style IMHO. Matters when dealing with objects since `c++` needs a copy to be stored and returned.

Changed to `++c` and made a quick sweep across the files to see if they are consistent.